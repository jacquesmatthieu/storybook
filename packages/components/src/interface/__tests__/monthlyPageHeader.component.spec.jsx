import React from 'react';
import { shallow } from 'enzyme';

import MonthlyPageHeader from '../monthlyPageHeader.component';

describe('MonthlyPageHeader', () => {
  let props;

  beforeEach(() => {
    props = {
      steps: ['foo', 'bar'],
      date: '2018-11-09',
      onChange: jest.fn(),
    };
  });

  const getWrapper = () => shallow(<MonthlyPageHeader {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle previous month click', () => {
    // Given
    const wrapper = getWrapper();

    // When
    wrapper.find('IconButton').first().simulate('click');

    // Then
    expect(props.onChange).toHaveBeenCalledWith('2018-10');
  });

  it('should handle next month click', () => {
    // Given
    const wrapper = getWrapper();

    // When
    wrapper.find('IconButton').last().simulate('click');

    // Then
    expect(props.onChange).toHaveBeenCalledWith('2018-12');
  });
});
