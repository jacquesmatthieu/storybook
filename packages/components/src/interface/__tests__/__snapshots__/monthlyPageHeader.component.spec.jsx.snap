// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyPageHeader should render properly 1`] = `
<monthlyPageHeadercomponent__Container
  className=""
>
  <WithCurrentApp(Breadcrumb)
    location={
      Object {
        "pathname": "foo",
      }
    }
    match={
      Object {
        "params": Object {},
      }
    }
    steps={
      Array [
        "foo",
        "bar",
      ]
    }
  />
  <monthlyPageHeadercomponent__MonthSelector>
    <IconButton
      data-test-id="previousMonthButton"
      hollow={false}
      icon="arrow"
      iconProps={
        Object {
          "direction": "left",
        }
      }
      onClick={[Function]}
    />
    <monthlyPageHeadercomponent__MonthSelectorText
      variant="headlineContainer"
    >
      November 2018
    </monthlyPageHeadercomponent__MonthSelectorText>
    <IconButton
      data-test-id="nextMonthButton"
      hollow={false}
      icon="arrow"
      iconProps={
        Object {
          "direction": "right",
        }
      }
      onClick={[Function]}
    />
  </monthlyPageHeadercomponent__MonthSelector>
</monthlyPageHeadercomponent__Container>
`;
